src/libs/actions/Report.ts

```diff
@@ -1498,6 +1498,64 @@ function toggleSubscribeToChildReport(childReportID = '0', parentReportAction: P
     }
 }
 
+function updatePolicyReportField(reportID: string, policyField: PolicyReportField, fieldValue: string) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                reportFields: {
+                    [policyField.fieldID]: fieldValue,
+                },
+                pendingFields: {
+                    [policyField.fieldID]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+            },
+        },
+    ];
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                pendingFields: {
+                    [policyField.fieldID]: null,
+                },
+                errorFields: {
+                    [policyField.fieldID]: ErrorUtils.getMicroSecondOnyxError('report.genericUpdateReportFieldFailureMessage'),
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                pendingFields: {
+                    [policyField.fieldID]: null,
+                },
+                errorFields: {
+                    [policyField.fieldID]: null,
+                },
+            },
+        },
+    ];
+
+    type UpdateReportFieldParameters = {
+        reportID: string;
+        reportFields: string;
+    };
+
+    const parameters: UpdateReportFieldParameters = {
+        reportID,
+        reportFields: JSON.stringify({[policyField.fieldID]: {fieldID: policyField.fieldID, value: fieldValue, type: policyField.type, name: policyField.name}}),
+    };
```

Comment:
Can we include all values from the policyField at this time? When OpenReport passes back the reports, it will have a reportField list that contains the reportFields at the time of creation. 

So if the weights change later, this report should have the same report fields weights at the time of storage for historical context